File: apps/api/app/schemas/knowledge.py
"""知识库 API Schema。"""

from pydantic import BaseModel


class KnowledgeBaseCreateRequest(BaseModel):
    actor_user_id: str
    org_id: str
    name: str
    description: str = ""


class KnowledgeBaseResponse(BaseModel):
    kb_id: str
    org_id: str
    name: str
    description: str
    created_by: str


class DocumentUploadRequest(BaseModel):
    actor_user_id: str
    title: str
    content: str
    chunk_size: int = 500
    chunk_overlap: int = 0


class DocumentResponse(BaseModel):
    document_id: str
    kb_id: str
    org_id: str
    title: str
    status: str
    created_by: str


class ChunkResponse(BaseModel):
    chunk_id: str
    document_id: str
    content: str
    sequence: int
    estimated_tokens: int
    embedding_model: str = ""
    vector_indexed: bool = False
    similarity_score: float | None = None


class SearchRequest(BaseModel):
    actor_user_id: str
    query: str
    limit: int = 5

File: apps/web/types/knowledge.ts
/** Knowledge 相关类型定义。 */

/** 知识库 */
export interface KnowledgeBaseItem {
  kb_id: string;
  org_id: string;
  name: string;
  description: string;
}

/** 文档 */
export interface DocumentItem {
  document_id: string;
  kb_id: string;
  title: string;
  status: string;
}

/** 文档块 */
export interface ChunkItem {
  chunk_id: string;
  document_id: string;
  content: string;
  sequence: number;
  estimated_tokens: number;
  embedding_model: string;
  vector_indexed: boolean;
  similarity_score: number | null;
}

/** 缓存统计 */
export interface CacheStats {
  size: number;
  max_size: number;
  total_hits: number;
  total_misses: number;
  hit_rate: number;
}

/** 知识库创建请求 */
export interface CreateKnowledgeBaseRequest {
  actor_user_id: string;
  org_id: string;
  name: string;
  description: string;
}
